gotopkg.go

```diff
@@ -1,204 +0,0 @@
-package main
-
-import (
-	"flag"
-	"fmt"
-	"github.com/groob/plist"
-	"io"
-	"io/ioutil"
-	"log"
-	"os"
-	"os/exec"
-	"strings"
-	"syscall"
-	"time"
-)
-
-const RECIPE_FAILED_CODE = 70
-
-type AutopkgRunReport struct {
-	Failures []struct {
-		Recipe    string `plist:"recipe"`
-		Message   string `plist:"message"`
-		Traceback string `plist:"traceback"`
-	} `plist:"failures"`
-	SummaryResults struct {
-		URLDownloaderSummaryResult struct {
-			DataRows []struct {
-				DownloadPath string `plist:"download_path"`
-			} `plist:"data_rows"`
-			Header      []string `plist:"header"`
-			SummaryText string   `plist:"summary_text"`
-		} `plist:"url_downloader_summary_result"`
-		MunkiImporterSummaryResult struct {
-			DataRows []struct {
-				PkgRepoPath string `plist:"pkg_repo_path"`
-				Catalogs    string `plist:"catalogs"`
-				Version     string `plist:"version"`
-				PkginfoPath string `plist:"pkginfo_path"`
-				Name        string `plist:"name"`
-			} `plist:"data_rows"`
-			Header      []string `plist:"header"`
-			SummaryText string   `plist:"summary_text"`
-		} `plist:"munki_importer_summary_result"`
-	} `plist:"summary_results"`
-}
-
-func (rpt *AutopkgRunReport) DecodePlist(r io.Reader) error {
-	err := plist.NewDecoder(r).Decode(&rpt)
-	if err != nil {
-		return fmt.Errorf("error decoding report from reader:", err)
-	}
-
-	return nil
-}
-
-var autopkg string
```

Comment:
Sure, in general I'm with you. The cost though is handing around a reference to your config option/global var to everything that needs it in a func/method parameter. But probably worth it.